src/scene/base.rs

```diff
@@ -973,11 +1073,58 @@ impl Visit for Base {
         //
         // None of the reasons are fatal and we should still give an ability to load such node
         // to edit or remove it.
-        if let Err(e) = visit_opt_script("Script", &mut self.script, &mut region) {
-            // Do not spam with error messages if there is missing `Script` field. It is ok
-            // for old scenes not to have script at all.
-            if !matches!(e, VisitError::RegionDoesNotExist(_)) {
-                Log::err(format!("Unable to visit script. Reason: {:?}", e))
+
+        if region.is_reading() {
```

Comment:
These two branches of `if` are way too convoluted. To preserve backward compatibility all you need to do is:

```rust
let old_script: Option<Script> = None;
if region.is_reading() && old_script.visit("Script", region).is_ok() {
  self.scripts.push(ScriptWrapper(old_script));
  return;
}

self.scripts.visit("Scripts", region)?;
```